chain/src/chain.rs

```diff
@@ -484,31 +490,36 @@ impl<CS: ChainStore + 'static> ChainService<CS> {
         let attached_blocks_iter = fork.attached_blocks().iter().rev();
         let detached_blocks_iter = fork.detached_blocks().iter().rev();
 
-        let unverified_len = fork.attached_blocks.len() - dirty_exts.len();
+        let mut fork_context = ForkContext {
+            attached_blocks: Vec::with_capacity(fork.attached_blocks().len()),
+            detached_blocks: fork.detached_blocks().iter().rev().collect(),
```

Comment:
Bad smell, similar structure, same field name, but have a different order.

Reply:
I unified the order. haven't idea about name, still .